doc/develop/test/twister_statuses.rst

```diff
@@ -0,0 +1,250 @@
+     - NO
+     - Suite, Case
+   * - STARTED
+     - started
+     - The test is currently in progress.
+     - NO
+     - Case
+   * - NOTRUN
+     - not run
+     - The test wasn't run because it was not runnable in current configuration.
+       It was, however, built correctly.
+     - YES
+     - Suite, Case
+   * - BLOCK
+     - blocked
+     - The test was not run because of an error or crash in the test suite.
+     - YES
+     - Case
+   * - FILTER
+     - filtered
+     - A static or runtime filter has eliminated the test from the list of tests to use.
+     - YES
+     - Suite, Case
+   * - ERROR
+     - error
+     - The test produced an error in running the test itself.
+     - YES
+     - Suite, Case
+   * - FAIL
+     - failed
+     - The test produced results different than expected.
+     - YES
+     - Suite, Case
+   * - PASS
+     - passed
+     - The test produced results as expected.
+     - YES
+     - Suite, Case
+   * - SKIP
+     - skipped
+     - Test was skipped by some other reason than previously delineated.
+     - YES
+     - Suite, Case
+
+**In-code** and **In-text** columns should the names of a given status.
+If one is not a Twister developer, the in-text name should suffice.
+It is the name used in the JSON reports.
+
+**Terminal** column indicates whether a given status should appear in the final Twister report.
+Appearance of non-terminal Statuses in one's report files indicates a problem with Twister.
+
+**Where applicable** column indicates where we might find such Statuses.
+Harness has been omitted for clarity, as it does not appear in final report files.
+If a status is marked as applicable for Suites, it is also applicable for Instances.
+
+
+Test Case and Test Suite Status combinations
+============================================
+
+.. list-table:: Test Case and Test Suite Status combinations
+   :widths: 16 12 12 12 12 12 12 12
+   :align: center
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Case\\Suite
+     - NONE
+     - FILTER
+     - ERROR
+     - FAIL
+     - PASS
+     - NOTRUN
+     - SKIP
+   * - NONE
+     - N/A
+     - N/A
+     - N/A
+     - N/A
+     - N/A
+     - N/A
+     - N/A
+   * - STARTED
+     - N/A
+     - N/A
+     - N/A
+     - N/A
+     - N/A
+     - N/A
+     - N/A
+   * - FILTER
+     - N/A
+     - ✓
+     - ✕
+     - ✕
+     - ✕
+     - ✕
+     - ✕
+   * - ERROR
+     - N/A
+     - ✕
+     - ✓
+     - ✕
+     - ✕
+     - ✕
+     - ✕
+   * - BLOCK
+     - N/A
+     - ✕
+     - ✓
+     - ✓
+     - ✕
+     - ✕
+     - ✕
+   * - FAIL
+     - N/A
+     - ✕
+     - ✓
+     - ✓
+     - ✕
+     - ✕
+     - ✕
+   * - PASS
+     - N/A
+     - ✕
+     - ✓
+     - ✓
+     - ✓
+     - ✕
+     - ✕
+   * - NOTRUN
+     - N/A
+     - ✕
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+     - ✕
+   * - SKIP
+     - N/A
+     - ✕
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+
+✕ indicates that such a combination should not happen in a proper Twister run. In other words,
```

Comment:
What is the meaning for 'N/A's in this table ?
IMO all of them could be excluded from here for simplicity, so the table might keep only 'V'-s where applicable.

Reply:
As mentioned above and below, those statuses are not terminal, so they should not appear at all. So N/A - not applicable.

This style of checkmarks, crosses and N/As is consistent with other tables - e.g. https://docs.zephyrproject.org/latest/services/device_mgmt/mcumgr.html

Reply:
> As mentioned above and below, those statuses are not terminal, so they should not appear at all. So N/A - not applicable.

These 'non-terminal' status' combinations are rather Twister internals and having them here filling the table with N/As is confusing, so it would be pretty enough just to have the note below that `.. their existence at the end of a Twister run indicates a problem with the Twister itself.`

Reply:
Added that information as a note and removed it from the tables.